Add support for pathlib

bburan opened the pull request.

mverleg commented on json_tricks/decoders.py:
```diff
@@ -117,6 +117,23 @@ def noenum_hook(dct):
 
 
+def pathlib_hook(dct):
+		if not isinstance(dct, dict):
```
This seems to be indented with double tabs, while the rest of the code is single tabs

bburan replied:
Sorry! I didn't realize the file was formatted with tabs when I first started editing, so I had to go back and convert everything from spaces to tabs. I'll fix this.

mverleg commented on json_tricks/nonp.py:
```diff
@@ -26,9 +26,9 @@
 _eih_instance = EnumInstanceHook()
 DEFAULT_ENCODERS = [json_date_time_encode, json_complex_encode, json_set_encode,
-                    numeric_types_encode, class_instance_encode, ]
+										numeric_types_encode, class_instance_encode, ]
```
Something went wrong with the indentation I think

mverleg commented on tests/test_pathlib.py:
```diff
@@ -0,0 +1,30 @@
+"""
+This tests timezone-aware date/time objects, which need pytz. Naive date/times should
+work with just Python code functionality, and are tested in `nonp`.
+"""
```
"This tests Paths, which need pathlib."

bburan replied:
Done.

mverleg commented on tests/test_pathlib.py:
```diff
@@ -0,0 +1,30 @@
+
+PATHS = [
+		Path(),
+		Path(__file__),
```
I think it's better to use a fixed path, so that the test is the same on each system

Maybe different OS paths too?

bburan replied:
Done.

mverleg commented:
Great addition, I wasn't aware of pathlib! And thanks for doing the implementation!

The changes are good, follow the setup of other encoders/decoders well.

There are just a few things left to do:

* A few mostly esthetic comments
* Include in readme (and contributors)
* Setup tox tests for pathlib

I could have a look this weekend to do the finishing touches, if you don't have time. I appreciate that you already did almost all the work :+1:

bburan commented:
I'm not an expert on tox, so I wasn't sure what to do. I updated the pathlib test to hide it inside an if `is_py3` block, but it looks like tox is smart enough to only run certain tests depending on the Python version? I think I addressed all the other comments you have, so please let me know if there's anything else. Thanks for taking the time to review this!

mverleg commented:
Okay looks great! I'll have a look at tox after merging, and add you to the contributors. Thanks for the PR!

mverleg commented:
This feature is available in the latest version through pip

bburan commented:
Thanks!

On Sat, Mar 2, 2019 at 3:31 PM Mark <notifications@github.com> wrote:

> This feature is available in the latest version through pip
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mverleg/pyjson_tricks/pull/54#issuecomment-468970642>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAT5QtuCqKwTyw8C3PfUAFWvXmM9uUnsks5vSwnbgaJpZM4bS9If>
> .
>
